openlibrary/utils/solr.py

```diff
@@ -77,6 +77,16 @@ def get_many(
         ).json()
         return [doc_wrapper(doc) for doc in resp['response']['docs']]
 
+    def update_in_place(
+        self,
+        request,
+    ):
+        resp = requests.post(
+            f'{self.base_url}/update?update.partial.requireInPlace=true,commit=true',
```

Comment:
`commit` is a bit of a dangerous option in solr land because if done too frequently it causes performance issues, let's make this as an argument that defaults to `False` and only set it true in the context where we need it.